src/inline-loading/inline-loading.component.ts

```diff
@@ -0,0 +1,87 @@
+import {
+	Component,
+	Input,
+	Output,
+	EventEmitter
+} from "@angular/core";
+
+@Component({
+	selector: "ibm-inline-loading",
+	template: `
+		<div class="bx--inline-loading">
+		<div class="bx--inline-loading__animation">
+			<div
+				*ngIf="success === false"
+				class="bx--loading bx--loading--small">
+				<svg class="bx--loading__svg" viewBox="-75 -75 150 150">
+					<circle cx="0" cy="0" r="37.5"></circle>
+				</svg>
+			</div>
+			<svg
+				*ngIf="success === true"
+				class="bx--inline-loading__checkmark-container bx--inline-loading__svg" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 10 10">
+				<polyline class="bx--inline-loading__checkmark" points="0.74 3.4 3.67 6.34 9.24 0.74"></polyline>
+			</svg>
+		</div>
+		<p *ngIf="success === false" class="bx--inline-loading__text">{{loadingText}}</p>
+		<p *ngIf="success === true" class="bx--inline-loading__text">{{successText}}</p>
+		</div>
+	`
+})
+export class InlineLoading {
+	/**
+	 * Specify the text description for the loading state.
+	 *
+	 * @memberof InlineLoading
+	 */
+	@Input() loadingText;
+	/**
+	 * Specify the text description for the success state.
+	 *
+	 * @memberof InlineLoading
+	 */
+	@Input() successText;
+
+	/**
+	 * Returns value `true` if the component is in the success state.
+	 * @readonly
+	 */
+	@Input() get success() {
+		return this._success;
+	}
+	/**
+	 * Set the component's state to match the parameter and emits onSuccess if it exits.
+	 */
+	set success (success: boolean) {
+		this._success = success;
+		if (this._success) {
```

Comment:
I see what the goal is, but this block will only emit true _after_ the value is set to true. The carbon usage docs say it should be set to true and emitted after 1.5s only if no callback is specified.

That said I don't think that pattern is useful in the angular world, so I'd just go with a straight `Input` and drop the EventEmitter and related code

Reply:
Well spotted!
My understanding is that the pattern drives user experience and is not there because of framework limitation.

If we're keeping this, the `if` statement should check against `success` and `this._success` should only be set in the `setTimeout` after delay has elapsed.